metadb/store_test.go

```diff
@@ -0,0 +1,70 @@
+package metadb
+
+import (
+	"context"
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWriteAndQuery(t *testing.T) {
+	// start the rqlite node
+	s := &service{
+		config: NewConfig(),
+		ready:  make(chan struct{}, 1),
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		// start the rqlite server
+		assert.Nil(t, s.Run(ctx), "run service")
+	}()
+	// wait until the rqlite node is ready
+	<-s.ready
+
+	assert.NotNil(t, s.db, "store is nil")
+
+	// create the table
+	creation := "CREATE TABLE foo (id INTEGER NOT NULL PRIMARY KEY, name TEXT)"
+	result, err := s.Write(ctx, creation)
+	assert.Nil(t, err, "execute creation statement")
+	assert.Empty(t, result.Error, "result after execution")
+
+	// insert the value to table
+	insertion := "insert into foo(id, name) values(1,'alon')"
+	result, err = s.Write(ctx, insertion)
+	assert.Nil(t, err, "execute insertion statement")
+	assert.Empty(t, result.Error, "result after execution")
+
+	// insert the value to table
+	insertion = "insert into foo(id, name) values(2,'belayu')"
+	result, err = s.Write(ctx, insertion)
+	assert.Nil(t, err, "execute insertion statement")
+	assert.Empty(t, result.Error, "result after execution")
+
+	// select a row from table with week level
+	selection := "select * from foo where id = 1"
+	rows, err := s.Query(ctx, selection, "weak")
+	assert.Nil(t, err, "query statements")
```

Comment:
It should be written with table driven tests [article](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) and [example](https://github.com/pastelnetwork/gonode/blob/master/common/errors/error_test.go)

Reply:
done